core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakePlanParser.scala

```diff
@@ -1,20 +1,68 @@
 package com.databricks.labs.remorph.parsers.snowflake
 
-import com.databricks.labs.remorph.parsers.PlanParser
+import com.databricks.labs.remorph.intermediate.{LineCommentNode, LogicalPlan, Origin, ParsingErrors, PlanGenerationFailure}
+import com.databricks.labs.remorph.parsers.{PlanParser, ProductionErrorCollector}
 import com.databricks.labs.remorph.parsers.snowflake.rules._
-import com.databricks.labs.remorph.{intermediate => ir}
-import org.antlr.v4.runtime.{CharStream, Lexer, ParserRuleContext, TokenStream}
+import com.databricks.labs.remorph.{BuildingAst, KoResult, Parsing, Transformation, WorkflowStage, intermediate => ir}
+import org.antlr.v4.runtime.tree.ParseTree
+import org.antlr.v4.runtime.{CharStreams, CommonTokenStream, Token}
 
-class SnowflakePlanParser extends PlanParser[SnowflakeParser] {
+import scala.util.control.NonFatal
+
+class SnowflakePlanParser extends PlanParser {
 
   private val vc = new SnowflakeVisitorCoordinator(SnowflakeParser.VOCABULARY, SnowflakeParser.ruleNames)
 
-  override protected def createLexer(input: CharStream): Lexer = new SnowflakeLexer(input)
-  override protected def createParser(stream: TokenStream): SnowflakeParser = new SnowflakeParser(stream)
-  override protected def createTree(parser: SnowflakeParser): ParserRuleContext = parser.snowflakeFile()
-  override protected def createPlan(tree: ParserRuleContext): ir.LogicalPlan = vc.astBuilder.visit(tree)
-  override protected def addErrorStrategy(parser: SnowflakeParser): Unit =
+  override def parseLogicalPlan(parsing: Parsing): Transformation[LogicalPlan] = {
```

Comment:
we need both the plan and the token stream to attach comments so had to move away from the original design

Reply:
> we need both the plan and the token stream to attach comments so had to move away from the original single-enity-transform design

Could we not just accumulate he comments in a structure that allows search by their location? Either after the lexer is called or just within the lexer directly? Then we would not need the token stream and this map or whatever it is can be retrieved from the common lexer, then passed to the generator. 

Perhaps we need to settle on a design for how we are resolving comment location before coding?